proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,545 @@
+  * Key: `source`
+  * URI: "authority"
+  * Remark:  We propose here the authority should be using the FQDN form such
+    as `icu4x.unicode.org`, as this may be more immediately recognizable due
+	to its prevalent use on the Internet.
+* Dimension
+  * Key: (a simple string)
+  * URI: path component such as `...locale_id/42/...`
+  * Remark: Every path component becomes a pair of path elements separated by
+  * a forward slash (`/`).
+
+For a multi-dimensional key, there may be several equivalent URI
+mappings, corresponding to different possible orderings of the
+dimensions and their values.  This allows us to omit any of the
+dimensions when specifying the URI, which can prove to be useful in
+query formulation.   It eliminates the guesswork, as every URI can be
+readily parsed into a key, and eliminates the need to introduce a
+hierarchical ordering in the URI, if such a hierarchy does not need to
+exist.
+
+> It is unclear at the moment if any of the encodings should be the
+"canonical" one.
+
+> We may need to either keep all dimensions flat.
+
+Example encodings are given below.
+
+```
+Key: {
+  source = "fuchsia.dev",
+  app="weirdy-birds",
+  message_id=42,
+  locale="en-US"
+}
+Value: "Hello, {user}"
+```
+
+### Example 1: Unicode data
+
+Always required dimensions are prefixed with "@" in the key.  Implementations
+may omit them in internal data structures only if they are otherwise available
+to the code and unambiguous.
+
+Key:
+
+```
+{
+  “@scheme” = “udata”
+  “@source” = “icu4x.unicode.org”
+  “langid” = “en-US”
+  “category” = “Decimal”,
+  “key” = “symbols-v1”,
+}
+```
+
+URI encoding:
+
+```
+udata://icu4x.unicode.org/langid/en-US/category/decimal/key/symbols-v1
+udata://icu4x.unicode.org/category/decimal/key/symbols-v1/langid/en-US
+udata://icu4x.unicode.org/key/symbols-v1/langid/en-US/category/decimal
+```
+
+> Any permutations of dimensions with the same values are equivalent.  
+
+> Perhaps we want to specify a canonical ordering of dimensions, for simpler comparison.
+
+### Example 2: Localization
+
+Key:
+
+```
+{
+  “@scheme” = “fuchsia-l10n”
+  “@source” = “fuchsia.dev”
+  “app” = “program-name”
+  “locale” = “en-US”
+  “message_id” = “42”
+}
+```
+
+
+URI encoding:
+```
+fuchsia-l10n://fuchsia.dev/app/program-name/locale/en-US/message-id/42
+```
+
+> “fuchsia-l10n” scheme means “localization, in Fuchsia style”.  “mozilla.org”
+> may define a different scheme that matches the localization process.
+
+> Other acceptable permutations are omitted for brevity.
+
+## JSON
+------------
+
+### Example 1: Unicode data
+
+Key:
+
+```
+{
+  “@scheme” = “udata”
+  “@source” = “icu4x.unicode.org”
+  “langid” = “en-US”
+  “category” = “Decimal”,
+  “key” = “symbols-v1”,
+}
+```
+
+JSON encoding:
+
+```
+{
+  “@scheme”: “udata”,
+  “@source”: “icu4x.unicode.org”,
+  “langid”:  “en-US”,
+  “category: “decimal”,
+  “key”:     “symbols-v1”
+}
+```
+
+> The notations ended up being very similar, as was probably expected.
```

Comment:
nit: unnecessary remark, probably just omit unless there is a specific message trying to be sent

Reply:
Done.